src/main/initVerticalMod.F90

```diff
@@ -257,24 +251,50 @@ subroutine initVertical(bounds, glc_behavior, snow_depth, thick_wall, thick_roof
 
        zisoi(0) = 0._r8
        do j = 1, nlevgrnd-1
-       zisoi(j) = 0.5_r8*(zsoi(j)+zsoi(j+1))         !interface depths
+          zisoi(j) = 0.5_r8*(zsoi(j)+zsoi(j+1))         !interface depths
        enddo
        zisoi(nlevgrnd) = zsoi(nlevgrnd) + 0.5_r8*dzsoi(nlevgrnd)
 
-    else if ( soil_layerstruct == '49SL_10m' ) then
-       !scs: 10 meter soil column, nlevsoi set to 49 in clm_varpar
-       do j = 1,10
-          dzsoi(j)= 1.e-2_r8     !10mm layers
-       enddo
-       do j = 11,19
-          dzsoi(j)= 1.e-1_r8     !100 mm layers
-       enddo
-       do j = 20,nlevsoi+1       !300 mm layers
-          dzsoi(j)= 3.e-1_r8
-       enddo
-       do j = nlevsoi+2,nlevgrnd !10 meter bedrock layers
-          dzsoi(j)= 10._r8
-       enddo
+    else if (calc_method == 'thickness-based') then
+       if (soil_layerstruct == '49SL_10m') then
+          !scs: 10 meter soil column, nlevsoi set to 49 in clm_varpar
+          do j = 1, 10
+             dzsoi(j) = 1.e-2_r8     ! 10-mm layers
+          enddo
+          do j = 11, 19
+             dzsoi(j) = 1.e-1_r8     ! 100-mm layers
+          enddo
+          do j = 20, nlevsoi+1       ! 300-mm layers
+             dzsoi(j) = 3.e-1_r8
+          enddo
+          do j = nlevsoi+2,nlevgrnd  ! 10-m bedrock layers
+             dzsoi(j) = 10._r8
+          enddo
+       else if (soil_layerstruct == '20SL_8.5m') then
+          do j = 1, 4  ! linear increase in layer thickness of...
+             dzsoi(j) = j * 0.02_r8                     ! ...2 cm each layer
+          enddo
+          do j = 5, 13
+             dzsoi(j) = dzsoi(4) + (j - 4) * 0.04_r8    ! ...4 cm each layer
+          enddo
+          do j = 14, nlevsoi
+             dzsoi(j) = dzsoi(13) + (j - 13) * 0.10_r8  ! ...10 cm each layer
+          enddo
+          do j = nlevsoi + 1, nlevgrnd  ! bedrock layers
+             dzsoi(j) = dzsoi(nlevsoi) + (((j - nlevsoi) * 25._r8)**1.5_r8) / 100._r8
+          enddo
+       else if (soil_layerstruct == '5SL_3m') then
+          dzsoi(1) = 0.1_r8
+          dzsoi(2) = 0.3_r8
+          dzsoi(3) = 0.6_r8
+          dzsoi(4) = 1.0_r8
+          dzsoi(5) = 1.0_r8
+       else if (soil_layerstruct(1:5) == 'user:') then
+          do j = 1, nlevgrnd
+             ! read string indices 8 to 9, 11 to 12, 14 to 15, and so on
+             read(soil_layerstruct(3*(j+2)-1:3*(j+2)),*) dzsoi(j)
```

Comment:
I need to change this to read 4 characters per dzsoi(j) since the units are meters and 4 characters will allow 
`min(dzsoi(j)) = .001 m`